js_modules/dagit/packages/core/src/ticks/EvaluateSchedule.tsx

```diff
@@ -0,0 +1,400 @@
+    variables: {
+      scheduleSelector: {
+        repositoryLocationName: repoAddress.location,
+        repositoryName: repoAddress.name,
+        scheduleName: name,
+      },
+    },
+  });
+  const [userTimezone] = React.useContext(TimezoneContext);
+  const [isTickSelectionOpen, setIsTickSelectionOpen] = React.useState<boolean>(false);
+  const selectedTimestampRef = React.useRef<{ts: number; label: string} | null>(null);
+  const {viewport, containerProps} = useViewport();
+  const [shouldEvaluate, setShouldEvaluate] = React.useState(false);
+  const content = React.useMemo(() => {
+    if (shouldEvaluate) {
+      return (
+        <EvaluateScheduleContent
+          repoAddress={repoAddress}
+          name={name}
+          timestamp={selectedTimestampRef.current!.ts}
+          jobName={jobName}
+        />
+      );
+    }
+    if (!data) {
+      return (
+        <Box padding={{vertical: 48}} flex={{alignItems: 'center', justifyContent: 'center'}}>
+          <Spinner purpose="page" />
+        </Box>
+      );
+    }
+    if (data.scheduleOrError.__typename === 'PythonError') {
+      return <div />;
+    }
+    if (data.scheduleOrError.__typename === 'ScheduleNotFoundError') {
+      return <div />;
+    }
+    const timestamps = data.scheduleOrError.potentialTickTimestamps.map((ts) => ({
+      ts,
+      label: timestampToString({
+        timestamp: {unix: ts},
+        locale,
+        timezone: userTimezone,
+        timeFormat: {
+          showTimezone: true,
+        },
+      }),
+    }));
+    selectedTimestampRef.current = _selectedTimestamp || timestamps[0];
+    return (
+      <SelectWrapper>
+        <Popover
+          isOpen={isTickSelectionOpen}
+          position="bottom-left"
+          fill={true}
+          content={
+            <Menu style={{maxHeight: '400px', overflow: 'scroll', width: `${viewport.width}px`}}>
+              {timestamps.map((timestamp) => (
+                <MenuItem
+                  key={timestamp.ts}
+                  text={<div data-testid={testId(`tick-${timestamp.ts}`)}>{timestamp.label}</div>}
+                  onClick={() => {
+                    setSelectedTimestamp(timestamp);
+                    setIsTickSelectionOpen(false);
+                  }}
+                />
+              ))}
+            </Menu>
+          }
+        >
+          <div {...containerProps}>
+            <Button
+              style={{flex: 1}}
+              rightIcon={<Icon name="arrow_drop_down" />}
+              onClick={() => setIsTickSelectionOpen((isOpen) => !isOpen)}
+              data-testid={testId('tick-selection')}
+            >
+              {selectedTimestampRef.current.label}
+            </Button>
+          </div>
+        </Popover>
+      </SelectWrapper>
+    );
+  }, [
+    _selectedTimestamp,
+    containerProps,
+    data,
+    isTickSelectionOpen,
+    jobName,
+    name,
+    repoAddress,
+    shouldEvaluate,
+    userTimezone,
+    viewport.width,
```

Comment:
I'm not sure this is worth memoizing -- with all these deps, it seems like this is probably being rebuilt on every render. What state isn't in this list? 😅